src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -291,9 +296,35 @@ public Response getResponse(String userName, String text) {
     // EEK! clean up the API!
     invoke("publishResponse", response);
     invoke("publishResponseText", response);
-    invoke("publishText", response.msg);
+    invoke("publishText", HtmlFilter.stripHtml(response.msg));

```

Comment:
Program AB should not be doing the HTML filtering here.  That is the job of the HtmlFilter service.

Reply:
ProgramAB by definition published HTML, do NOT strip it here!

Reply:
I agree AND disagree : we have 3 Publishers . we can give ourselves the luxury that one of them publishes only clean text. please :)

publishResponse > response
publishResponseText > response.msg
publishText > raw text filtered without html tags